app/models/user.rb

```diff
@@ -86,6 +86,39 @@ class User < ActiveRecord::Base
     },
     ranked_by: ":tsearch + (0.25 * :trigram)"
 
+  def self.dormant(window=5)
+    DatabaseReplica.read("read_dormant_users_from_replica") do
+      # devise trackable sets the current_sign_in_at on each login
+      havent_signed_in_since = "date(now()) - date(current_sign_in_at) >= #{window}"
+      where(havent_signed_in_since).select(:id).find_each do |dormant_user|
```

Comment:
This where clause should also respect the `receives_email?` states as described in https://github.com/zooniverse/Panoptes/blob/30e33a4f28b45269d2610ddf8eea0f8436ba40c6/app/models/user.rb#L201

We should skip disabled users, users without email attributes or ones that have been marked with invalid email addresses (bouncing). Doing this in the DB will filter out more users, it shouldn't impact on query times significantly but will require a quick benchmark on real data. Check in with Marten re this one.